### PR TITLE
SLLS-106 specify whether taint issue is coming from SQ or SC

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

File: src/test/java/org/sonarsource/sonarlint/ls/connected/events/ServerSentEventsTests.java [hunk 1]
```diff
@@ -398,4 +411,4 @@ private void mockFileInAFolder() {
         var folderWrapper = spy(new WorkspaceFolderWrapper(workspaceFolderPath.toUri(), new WorkspaceFolder(workspaceFolderPath.toUri().toString())));
         when(foldersManager.findFolderForFile(fileInAWorkspaceFolderPath.toUri())).thenReturn(Optional.of(folderWrapper));
     }
-}
+}
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/connected/events/ServerSentEventsTests.java:414]: Nitpick: missing new line at EOF

File: src/test/java/org/sonarsource/sonarlint/ls/connected/notifications/TaintVulnerabilityRaisedNotificationTests.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ void shouldCallShowTaintCommandWhenShowIssueActionSelected() {
     when(client.showMessageRequest(any())).thenReturn(CompletableFuture.completedFuture(SHOW_ISSUE_ACTION));
 
     taintVulnerabilityRaisedNotification.showTaintVulnerabilityNotification(fakeEvent, CONNECTION_ID, false);
+
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/connected/notifications/TaintVulnerabilityRaisedNotificationTests.java:73]: Nitpick: unnecessary diff 🙈

File: src/main/java/org/sonarsource/sonarlint/ls/connected/domain/TaintIssue.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return new TaintIssue(serverTaintIssue.getKey(), serverTaintIssue.isResolved(), serverTaintIssue.getRuleKey(),
+      serverTaintIssue.getMessage(), serverTaintIssue.getFilePath(), serverTaintIssue.getCreationDate(), serverTaintIssue.getSeverity(),
+      serverTaintIssue.getType(), serverTaintIssue.getTextRange(), source);
+  }
+
+  public static List<TaintIssue> from(List<ServerTaintIssue> serverTaintIssues, boolean isSonarCloudAlias) {
+    var source = isSonarCloudAlias ? AnalysisScheduler.SONARCLOUD_TAINT_SOURCE : AnalysisScheduler.SONARQUBE_TAINT_SOURCE;
+    List<TaintIssue> taintIssues = new ArrayList<>();
+    serverTaintIssues.forEach(serverTaintIssue -> taintIssues.add(TaintIssue.from(serverTaintIssue, source)));
+    return taintIssues;
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/connected/domain/TaintIssue.java:59]: Nitpick: could also be written as a stream + map + collect chain ⛓️

**Review by jblievremont — APPROVED** [review]
Minor nitpicks, else LGTM 👍🏻

**Review by kirill-knize-sonarsource — APPROVED** [review]
